backend/src/services/gitBookService/PageContentManager.ts

```diff
@@ -0,0 +1,50 @@
+import { RequirementStatusEnum } from "myTypes";
+
+type Requirement = {
+    status: RequirementStatusEnum;
+    requirement: string;
+};
+
+class GitBookPageContentManager {
+
+    constructor() {
+    }
+
+    extractRequirements(pageContent): Requirement[] {
+        if (!pageContent || !pageContent.document || !pageContent.document.nodes) {
+            console.log("Invalid page content format.");
+            return [];
```

Comment:
Is it expected behavior of this function? if not then we could throw error or return error list.

Reply:
If we have consecutive unrelated actions in higher function then returning error object instead of throwing one would be more beneficial.

Reply:
I will add a new error for the page content manager and ensure that if there is an error when processing any content page, it won't crash the processing. Instead, it will be independent, and other content pages can be fetched.